corgidrp/mocks.py

```diff
@@ -356,3 +359,146 @@ def create_badpixelmap_files(filedir=None, col_bp=None, row_bp=None):
+    new_hdr['CTYPE2'] = 'DEC--TAN'
+
+    new_hdr['CDELT1'] = (platescale * 0.001) / 3600
+    new_hdr['CDELT2'] = (platescale * 0.001) / 3600
+
+    new_hdr['CRVAL1'] = target[0]
+    new_hdr['CRVAL2'] = target[1]
+
+    w = wcs.WCS(new_hdr)
+
+    # create the image data
+    xpix, ypix = wcs.utils.skycoord_to_pixel(cal_SkyCoords, wcs=w)
+    xpix_inds = np.where((xpix >= 0) & (xpix <= 1024) & (ypix >= 0) & (ypix <= 1024))[0]
+    ypix_inds = np.where((ypix >= 0) & (ypix <= 1024) & (xpix >= 0) & (xpix <= 1024))[0]
+    xpix = xpix[xpix_inds]
+    ypix = ypix[ypix_inds]
+
+    amplitudes = np.power(10, ((cal_field['VMAG'] - 22.5) / (-2.5))) * 10
+
+    # inject gaussian psf stars
+    for xpos, ypos, amplitude in zip(xpix, ypix, amplitudes):
+        stampsize = int(np.ceil(3 * fwhm))
+        sigma = fwhm/ (2.*np.sqrt(2*np.log(2)))
+        
+        # coordinate system
+        y, x = np.indices([stampsize, stampsize])
+        y -= stampsize // 2
+        x -= stampsize // 2
+        
+        # find nearest pixel
+        x_int = int(round(xpos))
+        y_int = int(round(ypos))
+        x += x_int
+        y += y_int
+        
+        xmin = x[0][0]
+        xmax = x[-1][-1]
+        ymin = y[0][0]
+        ymax = y[-1][-1]
+        
+        psf = amplitude * np.exp(-((x - xpos)**2. + (y - ypos)**2.) / (2. * sigma**2))
+
+        # crop the edge of the injection at the edge of the image
+        if xmin <= 0:
+            psf = psf[:, -xmin:]
+            xmin = 0
+        if ymin <= 0:
+            psf = psf[-ymin:, :]
+            ymin = 0
+        if xmax >= nx:
+            psf = psf[:, :-(xmax-nx + 1)]
+            xmax = nx - 1
+        if ymax >= ny:
+            psf = psf[:-(ymax-ny + 1), :]
+            ymax = ny - 1
+
+        # inject the stars into the image
+        sim_data[ymin:ymax + 1, xmin:xmax + 1] += psf
+
+    # add Gaussian random noise
+    noise_rng = np.random.default_rng(10)
+    gain = 1
+    ref_flux = 10
+    noise = noise_rng.normal(scale= ref_flux/gain * 0.1, size= size)
+    sim_data = sim_data + noise
+
+    # load as an image object
+    frames = []
+    prihdr, exthdr = create_default_headers()
+    newhdr = fits.Header(new_hdr)
+    frame = data.Image(sim_data, pri_hdr= prihdr, ext_hdr= newhdr)
+    filename = "simcal_astrom.fits"
+    if filedir is not None:
+        # save source SkyCoord locations and pixel location estimates
+        guess = Table()
+        guess['x'] = [int(x) for x in xpix]
+        guess['y'] = [int(y) for y in ypix]
+        guess['RA'] = cal_SkyCoords[xpix_inds].ra
+        guess['DEC'] = cal_SkyCoords[ypix_inds].dec
+        ascii.write(guess, filedir+'/guesses.csv', overwrite=True)
+
+        center = Table()
+        center['RA'] = [target[0]]
+        center['DEC'] = [target[1]]
+        ascii.write(center, filedir+'/target_guess.csv', overwrite=True)
+
+        frame.save(filedir=filedir, filename=filename)
+
+    frames.append(frame)
+    dataset = data.Dataset(frames)
```

Comment:
is this a single image dataset? do we want to have the option to have more frames simulated?